napalm/nxos/nxos.py

```diff
@@ -595,7 +595,9 @@ def cli(self, commands):
             cli_output[py23_compat.text_type(command)] = command_output
         return cli_output
 
-    def get_arp_table(self):
+    def get_arp_table(self, vrf=c.ARP_VRF):
+        if vrf:
```

Comment:
We should probably use a helper method for this. Something like:

```
# in something something helpers.py
def raise_if_not_supported(supported, args):
    "if an argument in args is not in supported raise NotImplementedError" 
   for kw in args:
       if kw not in supported:
           raise NotImplementedError("{} not implemented".format(kw))

# in drivers
def get_arp_table(self, **kwargs):
    helpers.raise_if_not_supported(['vrf'], kwargs.keys())
```

With this pattern we can start migrating all the existent methods to it so when we implement a new arg we don't have to care about it. We could also add an optional arg "ignore_not_implemented_kwargs" or something like that avoid raising a NotImplementedError if the user passes a kwargs that is not supported.

I don't like it, btw, but would simplify implementing new kwargs and even vendor specific options like commit messages.